src/resources/js/events-admin.js

```diff
@@ -285,7 +285,8 @@ jQuery( document ).ready( function( $ ) {
 			if ( dropdown.length ) {
 				var value = dropdown.val();
 				if ( 0 !== parseInt( value, 10 ) ) {
-					fields.hide();
+					//hide all fields, but those with not-linked class i.e. Google Map Settings
+					fields.not('.not-linked').hide();
```

Comment:
Add some spaces in the function call as per [WordPress JS coding standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/javascript/#arrays-and-function-calls):

```js
fields.not( '.not-linked' ).hide();
```